pkg/csi/service/fcd/controller.go

```diff
@@ -224,15 +225,101 @@ func (c *controller) ControllerPublishVolume(
 	req *csi.ControllerPublishVolumeRequest) (
 	*csi.ControllerPublishVolumeResponse, error) {
 
-	return nil, nil
+	//check for required parameters
+	if len(req.VolumeId) == 0 {
+		msg := fmt.Sprintf("Volume ID is a required parameter.")
```

Comment:
no need for `Sprintf` here or below.  can also s/Errorf/Error/

Reply:
yea... it is just from copying an pasting to make it easier.